### PR TITLE
[Microsoft] Store resourcePath and standardize internalIds

### PR DESCRIPTION
**Pull request by philipperolet**
## Description
This PR
- stores the path of a resource, so that given a row from MicrosoftRootModel we have the data to query it via microsoft's graph api;
- standardize internal ids: 
  - add a provider suffix like for other connectors
   - ensure that we can compute internal id from row, and vice versa that given solely the internal id we can also query the api

For review: best  to start by L111 of `graph_api.ts` introducing MicrosoftNodeData <-> InternalId conversion functions

Also starts the code of the full-sync activity (still a no-op at the time)

## Risk & deploy
na (gated)

### CODE REVIEW

File: connectors/src/lib/models/microsoft.ts [hunk 1]
```diff
@@ -83,6 +88,10 @@ MicrosoftRootModel.init(
       type: DataTypes.INTEGER,
       allowNull: false,
     },
+    resourcePath: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
```
Comment (tdraier) [inline connectors/src/lib/models/microsoft.ts:94]: Didn't we say we want to avoid using `resource` for this  ? can it be `nodePath` ?

Comment (philipperolet) [reply connectors/src/lib/models/microsoft.ts]: changed to `itemApiPath`

**Comment by philipperolet** [comment]
As discussed IRL, we also delete `nodeId` from the model, already contained in `itemApiPath` (but not carrying enough info by itself)

**Comment by tdraier** [comment]
Approved !

**Pull request merged**